.github/workflows/ci-macos.yml

```diff
@@ -24,23 +24,27 @@ jobs:
     - uses: actions/setup-python@v5
       with:
         python-version: '3.11'
-        cache: 'pip' # caching pip dependencies
+        cache: 'pip'
+        cache-dependency-path: '**/requirements.txt'
```

Comment:
Do you have to provide this auxillary file? Or can the three dependencies just be listed in this file somehow?

Reply:
It has to be provided in order to cache the dependencies (described here): https://github.com/actions/setup-python/tree/main?tab=readme-ov-file#caching-packages-dependencies